### PR TITLE
Show Team Member Story Point capacity

### PR DESCRIPTION
**Pull request by manstis**
Add a new column showing a Team Members "Story Point" capacity.

This is based on their availability as a proportion of the whole teams availability.

i.e. A Team member who is only available for the half the time of another is only _capable_ of 1/3 of the whole team capacity.

image

Team Members 1 is available for twice the time of Team Members 2 and 3.

Therefore they potentially have the capacity to do twice as much as Team Members 2 and 3.

This can be used as a guide when allocating JIRA to people in Sprint Planning. If a Sprint shows Team Member 1 having a capacity of 20 Story Points we should ask questions if their assigned JIRA total exceeds this.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -212,17 +213,18 @@ <h4>Powered By <a href="https://github.com/omaciel/capacitator" target="_blank"
...
             const cell2 = newRow.insertCell(1);
             const cell3 = newRow.insertCell(2);
-            
+            const cell4 = newRow.insertCell(3);
+
             // Add input fields in the new row
-            cell1.innerHTML = '<input type="text" class="form-control" placeholder="Teammate Name">';
-            cell2.innerHTML = `<input type="number" class="form-control available-days" value="${document.getElementById('sprintDays').innerText}" oninput="calculateTotalTeamDays(); updateTableRowColor(this.parentElement.parentElement);">`;
-            cell3.innerHTML = '<input type="text" class="form-control" placeholder="">';
+            cell1.innerHTML = '<input type="text" id="teammate-name" class="form-control" placeholder="Teammate Name">';
```
Comment (manstis) [inline index.html:219]: Added `id`'s (as Chrome's DevTools was reporting them missing).

File: index.html [hunk 2]
```diff
@@ -212,17 +213,18 @@ <h4>Powered By <a href="https://github.com/omaciel/capacitator" target="_blank"
...
+            cell3.innerHTML = '<input type="text" id="teammate-pto" class="form-control" placeholder="">';
+            cell4.innerHTML = '<input type="number" id="teammate-capacity" class="form-control" value="0" disabled="disabled">';
 
-            const cell4 = newRow.insertCell(3);
-            cell4.innerHTML = '<button class="btn btn-danger btn-sm" onclick="deleteTeamMember(this)"><i class="bi bi-dash-circle"></i> Delete</button>';
+            const cell5 = newRow.insertCell(4);
+            cell5.innerHTML = '<button class="btn btn-danger btn-sm" onclick="deleteTeamMember(this)"><i class="bi bi-dash-circle"></i> Delete</button>';
 
             updateTeamSize();
-            calculateTotalTeamDays();
```
Comment (manstis) [inline index.html:225]: This wasn't needed as `updateTeamSize()` calls it.

File: index.html [hunk 3]
```diff
@@ -389,6 +415,7 @@ <h4>Powered By <a href="https://github.com/omaciel/capacitator" target="_blank"
                     updateTeamSize();
                     calculateSprintDays()
                     calculateTotalTeamDays();
+                    updateTeamMembersStoryPointCapacity();
```
Comment (manstis) [inline index.html:418]: This is in the `loadTemplateFromFile()` function (although this `diff` suggests it's in `saveTemplateAsJSON()`).

I don't store the `capacity` field in the export file and it is recalculated when loaded.

File: index.html [hunk 4]
```diff
@@ -321,6 +324,29 @@ <h4>Powered By <a href="https://github.com/omaciel/capacitator" target="_blank"
...
+            const rows = document.querySelectorAll("#teamTable tr");
+            rows.forEach(row => {
+                const teamMemberDays = parseFloat(row.cells[1].querySelector("input").value);
+                const teamMemberCapacity = Math.round(teamMemberDays / totalTeamDays * targetStoryPoints);
+                row.cells[3].innerText = teamMemberCapacity.toString();
+                totalTeamMembersCapacity += teamMemberCapacity;
+            });
+
+            // Fudge a random Team member's capacity so the totals match
+            if(totalTeamMembersCapacity !== targetStoryPoints) {
```
Comment (omaciel) [inline index.html:343]: I like this bit the best :D

**Pull request merged**